README.md

```diff
@@ -128,15 +165,36 @@ OakMap<K,V> oak = builder.build();
-
-For backward compatibility with applications that are already based on the use of ByteBuffers, Oak Buffers provide the transform method that atomically applies a transformation to a *read-only* instance of the underlying ByteBuffer. For a more comprehensive code example please refer to the [usage](#usage) section. 
+### Oak Buffers
+Oak use dedicated buffer objects to access off-heap memory. 
+These buffers cannot be instantiated by the user and are always supplied to the user by Oak.
+Their interfaces are:
+ 1. `OakReadBuffer`: scoped, read-only access to the data
+ 2. `OakWriteBuffer`: scoped, read and write access to the data
+ 3. `OakScopeFreeBuffer`: scope-free, read-only access to the data
+
+These buffers may represent either a key or a value.
+They mimic the standard interface of Java's `ByteBuffer`, for example, `int getInt(int index)`, `char getChar(int index)`, `capacity()`, etc. 
+
+Scoped buffers (`OakReadBuffer` and `OakWriteBuffer`) can only be used in the scope of the callback method they were first introduced to the user. 
+Such callback method might be the application's serializer and comparator, or a lambda function that can read/store/update the data (see [full API](https://github.com/yahoo/Oak/wiki/Full-API) for more information).
+This access reduces unnecessary copies and deserialization of the underlying data.
+In their intended context, the user does not need to worry about concurrent accesses and memory management.
+Using these buffers outside their intended context may yield unpredicted results, e.g., reading non-consistent data and/or irrelevant data.
+
+The third buffer (`OakScopeFreeBuffer`) is scope free, i.e., its usage is not scope limited and may be stored for future use.
+Some OakMap's methods returns this buffer to avoid copying the data and instead the user can access to the underlying memory buffer directly (lazy evaluation).
+While the scoped buffers are mutually exclusive, when using `OakScopeFreeBuffer`, the same memory might be access by a concurrent update operations.
+Thus, the reader may encounter different values -- and even value deletions -- when accessing `OakScopeFreeBuffer` multiple times.
+Specifically, when trying to access a deleted mapping via a `OakScopeFreeBuffer`, `ConcurrentModificationException` will be thrown.
+This is of course normal behavior for a concurrent map that avoids copying.
+To allow complex, multi-value atomic operations on the data, `OakScopeFreeBuffer` provides a `transform()` method that allow the user to apply a transformation function atomically on a read-only, scoped version of the buffer (`OakReadBuffer`). 
+See the [zero-copy API](#zero-copy-api) for more information.
+
+For backward compatibility with applications that are already based on the use of `ByteBuffer`, Oak's buffers also implements a dedicated unsafe interface `OakUnsafeDirectBuffer`.
```

Comment:
More elaborate explanations on OakUnsafeDirectBuffer are required here or in additional section. The *full API* doesn't explain it, *full API* are just methods. Even if it would something need to be here as well.

Reply:
fixed